olive/evaluator/olive_evaluator.py

```diff
@@ -156,22 +156,23 @@ def get_user_config(metric: Metric):
         eval_func = user_module.load_object(evaluate_func)
```

Comment:
It depends on the model and data. 

for the user_config part, it is used to create dummy data. 
for snpe_model, we need to get it from the model itself since we don't know the model's io config before hand. SNPE permutes the input shapes from `batchXchannelXwidthXheight` to `batchXwidthXheightXchannel` so we need both the original shapes (from data) and the model shape.

Reply:
for onnx model, we cannot use it to create dummy data so the latency metric user config might have io config inside them.

Reply:
I think we could revisit these later if needed.

Reply:
@jiapeng I think there are some plans to maybe remove `user_config` method of creating dataloaders in the future? Maybe we can review this then. Since the user_config `input_names`, `output_names` way of creating dummy data is already covered by the `DummyDataContainer`.

Reply:
Sounds good to me currently. Thanks.

In the future, maybe we can consider to let io_config spread to onnx model as well?
 Or another method for dynamic axis, we can assign the value if user does not provide it. Just make sure it is consist during the comparation in perf_tuning?

Reply:
Auto assigning would work too but I don't know if the perf tuning results would be as useful as using user provided shapes. Since we want to tune for data the user expects to use the model irl.

Reply:
Good catch. The real case will be more flexible. The auto io_config assigning will work for perf_tuning accurately only when user know we will use the io_config in input model for perf tuning as well.

Reply:
Okay. I will close this conversation for now since it is related to a different feature.